contracts/StableJoeStaking.sol

```diff
@@ -131,16 +135,19 @@ contract StableJoeStaking is Initializable, OwnableUpgradeable {
             IERC20Upgradeable _token = rewardTokens[i];
             updateReward(_token);
 
+            uint256 _previousRewardDebt = user.rewardDebt[_token];
+            user.rewardDebt[_token] = _newAmount.mul(accRewardPerShare[_token]).div(ACC_REWARD_PER_SHARE_PRECISION);
```

Comment:
this change is to make sure we follow checks-effects-interaction pattern before we perform the `safeTokenTransfer` in line 147